gateway/gateway.go

```diff
@@ -469,6 +479,23 @@ func (g *gateway) DoCommand(ctx context.Context, cmd map[string]interface{}) (ma
 	return map[string]interface{}{}, nil
 }
 
+// Criteria to determine if packets are identical:
+//
+//	-- count_us should be equal or can have up to 24µs of difference (3 samples)
+//	-- freq should be same
+//	-- datarate should be same
+//	-- payload should be same
+func isSamePacket(p1, p2 C.struct_lgw_pkt_rx_s) bool {
+	if math.Abs(float64(p1.count_us-p2.count_us)) <= 24 &&
+		p1.freq_hz == p2.freq_hz &&
+		p1.datarate == p2.datarate &&
+		//nolint:gocritic
+		C.memcmp(unsafe.Pointer(&p1.payload[0]), unsafe.Pointer(&p2.payload[0]), C.size_t(len(p1.payload))) == 0 {
+		return true
```

Comment:
We should isolate C calls more stringently from the go code, pleas make a ticket fro cleanup/test coverage of Cgo code.